pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -350,37 +314,93 @@ func (r *ReconcilePerconaServerMongoDB) handleReplsetInit(m *api.PerconaServerMo
 
 		time.Sleep(time.Second * 5)
 
-		cmd[2] = fmt.Sprintf(
-			`
-			cat <<-EOF | mongo 
-			%s
-			EOF
-			`, mongoInitAdminUser)
-		errb.Reset()
-		outb.Reset()
-		err = r.clientcmd.Exec(&pod, "mongod", cmd, nil, &outb, &errb, false)
+		userAdmin, err := r.getInternalCredentials(m, roleUserAdmin)
 		if err != nil {
-			return fmt.Errorf("exec add admin user: %v / %s / %s", err, outb.String(), errb.String())
+			return errors.Wrap(err, "failed to get userAdmin credentials")
 		}
 
-		cmd[2] = fmt.Sprintf(
-			`
-			cat <<-EOF | mongo "mongodb://${MONGODB_USER_ADMIN_USER}:${MONGODB_USER_ADMIN_PASSWORD}@%s/?replicaSet=%s"
+		cmd[2] = fmt.Sprintf(`
+			cat <<-EOF | mongo 
 			%s
-			EOF
-			`, host, replset.Name, mongoInitUsers)
+			EOF`, mongoInitAdminUser(userAdmin.Username, userAdmin.Password))
 		errb.Reset()
 		outb.Reset()
 		err = r.clientcmd.Exec(&pod, "mongod", cmd, nil, &outb, &errb, false)
 		if err != nil {
-			return fmt.Errorf("exec add users: %v / %s / %s", err, outb.String(), errb.String())
+			return fmt.Errorf("exec add admin user: %v / %s / %s", err, outb.String(), errb.String())
 		}
 
+		log.Info("replset was initialized", "replset", replset.Name, "pod", pod.Name)
+
 		return nil
 	}
+
 	return errNoRunningMongodContainers
 }
 
+func (r *ReconcilePerconaServerMongoDB) createSystemUsers(cr *api.PerconaServerMongoDB, replset *api.ReplsetSpec, pods []corev1.Pod) error {
+	cli, err := r.mongoClientWithRole(cr, replset.Name, replset.Expose.Enabled, pods, roleUserAdmin)
+	if err != nil {
+		return errors.Wrap(err, "failed to get mongo client")
+	}
+
+	defer func() {
+		err := cli.Disconnect(context.TODO())
+		if err != nil {
+			log.Error(err, "failed to close mongo connection")
+		}
+	}()
+
+	clusterAdmin, err := r.getInternalCredentials(cr, roleClusterAdmin)
+	if err != nil {
+		return errors.Wrap(err, "failed to get cluster admin")
+	}
+
+	err = mongo.CreateUser(context.Background(), cli, clusterAdmin.Username, clusterAdmin.Password, []interface{}{"clusterAdmin"})
+	if err != nil {
+		return errors.Wrap(err, "failed to create clusterAdmin")
+	}
+
+	monitorUser, err := r.getInternalCredentials(cr, roleMonitorUser)
+	if err != nil {
+		return errors.Wrap(err, "failed to get cluster admin")
+	}
+
+	err = mongo.CreateUser(context.Background(), cli, monitorUser.Username, monitorUser.Password, []interface{}{"clusterMonitor"})
```

Comment:
can we use const `roleMonitorUser`

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/548/commits/681651964d009c7ec4b0b95c1a4a7d902f4633ef